src/text-editor-component.js

```diff
@@ -4009,6 +3923,90 @@ class LineComponent {
   }
 }
 
+class HighlightsComponent {
+  constructor (props) {
+    this.props = {}
+    this.element = document.createElement('div')
+    this.element.className = 'highlights'
+    this.element.style.contain = 'strict'
```

Comment:
@nathansobo: I tentatively set the layout containment to `strict` now that highlights are not inside tiles anymore. Previously, we were using only layout containment (which, as a side note, I don't think was being enforced by Chromium, because we wrote the property name and the value in the opposite order, e.g. `layout: contain` instead of `contain: layout`). 

Do you have any concerns about switching to a stricter form of containment? Rendering-wise everything seems to displayed correctly, but I'd love to have your input on this.

Reply:
If everything is working well it seems reasonable to apply strict layout containment. We can't apply it to the line tiles because we want to allow text to escape them at awkward line heights and the highlights need to show through from behind, correct?